File: mirai-core/src/main/java/net/mamoe/mirai/event/events/network/PacketEvent.java
package net.mamoe.mirai.event.events.network;

import net.mamoe.mirai.Bot;
import net.mamoe.mirai.event.events.bot.BotEvent;
import net.mamoe.mirai.network.packet.Packet;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * @author Him188moe
 */
public abstract class PacketEvent extends BotEvent {
    private final Packet packet;

    public PacketEvent(@NotNull Bot bot, @NotNull Packet packet) {
        super(bot);
        this.packet = Objects.requireNonNull(packet);
    }

    public Packet getPacket() {
        return packet;
    }
}

File: mirai-core/src/jvmMain/kotlin/net/mamoe/mirai/event/events/network/PacketEvent.kt
package net.mamoe.mirai.event.events.network

import net.mamoe.mirai.Bot
import net.mamoe.mirai.event.events.bot.BotEvent
import net.mamoe.mirai.network.protocol.tim.packet.Packet

/**
 * @author Him188moe
 */
abstract class PacketEvent<out P : Packet>(bot: Bot, open val packet: P) : BotEvent(bot)
